test/spell/RelinkV2Test.sol

```diff
@@ -0,0 +1,269 @@
+
+import {PoolId} from "../../src/common/types/PoolId.sol";
+import {AssetId} from "../../src/common/types/AssetId.sol";
+import {ShareClassId} from "../../src/common/types/ShareClassId.sol";
+import {ISpokeGatewayHandler} from "../../src/common/interfaces/IGatewayHandlers.sol";
+
+import {IShareToken} from "../../src/spoke/interfaces/IShareToken.sol";
+
+import {IBaseVault} from "../../src/vaults/interfaces/IBaseVault.sol";
+
+import {
+    UpdateRestrictionType,
+    UpdateRestrictionMessageLib
+} from "../../src/hooks/libraries/UpdateRestrictionMessageLib.sol";
+
+import "forge-std/Test.sol";
+
+contract RelinkV2TestBase is Test {
+    uint256 public constant REQUEST_ID = 0;
+
+    RelinkV2Eth public spell;
+
+    function setUp() public {
+        vm.createSelectFork("https://ethereum-rpc.publicnode.com");
+
+        spell = new RelinkV2Eth();
+    }
+
+    function castSpell() internal {
+        // NOTE: In production, this would be done through Guardian.scheduleRely + ROOT.executeScheduledRely
+        // For testing, we simulate this by directly setting the spell as ward on ROOT
+        // wards mapping is at slot 0 for Auth contracts
+        bytes32 wardSlot = keccak256(abi.encode(address(spell), uint256(0)));
+        vm.store(address(spell.V2_ROOT()), wardSlot, bytes32(uint256(1)));
+
+        spell.cast();
+    }
+}
+
+contract RelinkV2TestIntegrity is RelinkV2TestBase {
+    uint256 preJaaaTotalSupply;
+
+    // Main integration test
+    function testSpellIntegration() public {
+        preJaaaTotalSupply = spell.JAAA_SHARE_TOKEN().totalSupply();
+
+        castSpell();
+
+        _checkLinkedVaults();
+        _checkShareBalances();
+        _checkInvestmentState();
+        _checkMembershipState();
+        _checkAxelarTransactionCannotBeReexecuted();
+        _checkSpellCannotBeCastedSecondTime();
+        _checkSpellAccessIsCleared();
+    }
+
+    function _checkLinkedVaults() internal view {
+        assertEq(
+            spell.JTRSY_SHARE_TOKEN().vault(spell.USDC_TOKEN()),
+            spell.JTRSY_VAULT_ADDRESS(),
+            "JTRSY_SHARE_TOKEN.vault mismatch"
+        );
+        assertEq(
+            spell.JAAA_SHARE_TOKEN().vault(spell.USDC_TOKEN()),
+            spell.JAAA_VAULT_ADDRESS(),
+            "JAAA_SHARE_TOKEN.vault mismatch"
+        );
+    }
+
+    function _checkShareBalances() internal view {
+        uint256 postJaaaTotalSupply = spell.JAAA_SHARE_TOKEN().totalSupply();
+        assertEq(preJaaaTotalSupply, postJaaaTotalSupply);
+
+        assertEq(spell.JAAA_SHARE_TOKEN().balanceOf(spell.JAAA_INVESTOR()), 50_000_000e6);
+        assertEq(spell.JAAA_SHARE_TOKEN().balanceOf(address(spell)), 0);
+    }
+
+    function _checkInvestmentState() internal view {
+        VaultLike vault = VaultLike(spell.JAAA_VAULT_ADDRESS());
+        assertEq(vault.pendingDepositRequest(REQUEST_ID, spell.JAAA_INVESTOR()), 0);
+        assertEq(vault.claimableDepositRequest(REQUEST_ID, spell.JAAA_INVESTOR()), 0);
+    }
```

Comment:
Should we maybe check against the whole state of the `Investments` struct? 

<img width="1270" height="247" alt="image" src="https://github.com/user-attachments/assets/e47a98b8-c8ae-405d-b7af-ff6719147e6d" />